### PR TITLE
Release v0.12.2

### PR DESCRIPTION
**Pull request by hariso**
Part of https://github.com/ConduitIO/conduit/issues/1950.

### CODE REVIEW

File: changelog/2024-11-08-conduit-0-12-2-release.mdx [hunk 1]
```diff
@@ -0,0 +1,34 @@
+---
+slug: '2024-11-08-conduit-0-12-2-release'
+title: Conduit v0.12.2 release
+draft: false
+tags: [conduit, performance]
+---
+
+v0.12.2 comes with up to 5x more throughput, new CLI command for initializing
```
Comment (raulb) [inline changelog/2024-11-08-conduit-0-12-2-release.mdx:8]:
```suggestion
Conduit [v0.12.2](https://github.com/ConduitIO/conduit/releases/tag/v0.12.2) comes with up to 5x more throughput, new CLI command for initializing
```

File: changelog/2024-11-08-conduit-0-12-2-release.mdx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+Conduit and pipelines, as well as bug fixes.
+
+<!--truncate-->
+
+### Key Highlights
+
+- **Up to 5x higher throughput**: We started revamping the internal engine
+  that's running pipelines which resulted in a 5x higher throughput. If you're
+  interested in how we did it, you can find more information
+  in [the pull request](https://github.com/ConduitIO/conduit/pull/1913). We'll
```
Comment (raulb) [inline changelog/2024-11-08-conduit-0-12-2-release.mdx:18]:
```suggestion
  in [this pull request](https://github.com/ConduitIO/conduit/pull/1913). We'll
```

File: changelog/2024-11-08-conduit-0-12-2-release.mdx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  also publish a blog post where we will go into the details. The feature is
+  still experimental, and can be enabled with the `-preview.pipeline-arch-v2`
```
Comment (raulb) [inline changelog/2024-11-08-conduit-0-12-2-release.mdx:20]: Here i would include how running the entire command to better illustrate it:

```bash
$ conduit -preview.pipeline-arch-v2
```

I'm also wondering if we should have this buried in a changelog or we should highlight it elsewhere (even while it's experimental). Somewhere we could link people to. What do you think @hariso ?

Comment (raulb) [reply changelog/2024-11-08-conduit-0-12-2-release.mdx:20]: We should also include this flag in `/docs/configuration`.

**Review by raulb — CHANGES_REQUESTED** [review]
Some small suggestions. We also need to update the banner in `docusaurus.config.ts` linking to this new changelog.

```
announcementBar: {
      id: 'announcement-bar-6', // increment on change
      content: `Conduit v0.12.2 is here! <a class='cta' href='https://conduit.io/changelog/2024-11-08-conduit-0-12-2-release' target='_blank' rel='noreferrer noopener'>See what's new</a>.`,
      isCloseable: true,
    },
```

**Comment by raulb** [comment]
@hariso forgot to mention. We need to also update generally the output of `conduit help` to include `init`.

**Comment by hariso** [comment]
> Some small suggestions. We also need to update the banner in `docusaurus.config.ts` linking to this new changelog.
> 
> ```
> announcementBar: {
>       id: 'announcement-bar-6', // increment on change
>       content: `Conduit v0.12.2 is here! <a class='cta' href='https://conduit.io/changelog/2024-11-08-conduit-0-12-2-release' target='_blank' rel='noreferrer noopener'>See what's new</a>.`,
>       isCloseable: true,
>     },
> ```

I wasn't sure if we should be doing this for patch releases, that's why I skipped. I guess nor harm in announcing it this way too.:)

File: changelog/2024-11-08-conduit-0-12-2-release.mdx (hunk 3 above)
Comment (hariso) [reply changelog/2024-11-08-conduit-0-12-2-release.mdx]: I have added the full command and updated the configuration page, good callout! 

As for highlighting the performance stuff: given that it's a config param affecting performance, we can either mention it in the configuration page (which we just did) or on a performance optimization page (which we currently don't have). Given that it's still experimental, I think it's not bad that we leave it on the configuration page.

**Review by raulb — APPROVED** [review]

**Pull request merged**